File: Cpp/ride.cpp
/*
ID: chaoswo1
TASK: ride
LANG: C++
*/
#define CHAOSWORK
#define _CRT_SECURE_NO_WARNINGS

#include<iostream>
#include<vector>
#include<queue>
#include<cmath>
#include<cstdio>
#include<cstdlib>
#include<cstring>
#include<string>
#include<map>
#include<set>
#include<queue>
#include<algorithm>
using namespace std;

int getMod(const string& str)
{
	int i;
	int res=1;
	for(i=0;i<str.size();++i)
	{
		res*=(str[i]-'A')+1;
		res%=47;
	}
	return res;
}

int main()
{
    freopen("ride.in","r",stdin);
    freopen("ride.out","w",stdout);
	string comet;
	string group;
	cin>>comet;
	cin>>group;
	if(getMod(comet)!=getMod(group))
		cout<<"STAY"<<endl;
	else cout<<"GO"<<endl;
	
    return 0;
}

File: Java/ride.java


/*
ID: chaoswo1
LANG: JAVA
TASK: ride
*/
import java.io.*;
import java.util.*;

public class ride {
	
	static int getMod(String str){
		
		int res=1;
		for(int i=0;i<str.length();++i){
			res*=str.charAt(i)-'A'+1;
			res%=47;
			System.out.println(res);
		}
		System.out.println();
		return res;		
	}
	public static void main (String [] args) throws IOException {
		// Use BufferedReader rather than RandomAccessFile; it's much faster
		BufferedReader f = new BufferedReader(new FileReader("ride.in"));
		// input file name goes above
		PrintWriter out = new PrintWriter(new BufferedWriter(new FileWriter("ride.out")));
		
		String comet=f.readLine();
		String group=f.readLine();
		
		if(getMod(comet)!=getMod(group)){
			out.println("STAY");
		}
		else out.println("GO");
		
		out.close();
		System.exit(0);                               // don't omit this!
  }
}

